### PR TITLE
Avoid sql error when importing a file with the wrong field delimiter

### PR DESCRIPTION
**Pull request by bramley**
## Description

This forum topic  https://discuss.phplist.org/t/attributes-not-working-on-import/6768 showed an sql error when trying to import a comma separated file but not changing the field separator from TAB. This change escapes the column name when trying to match it with an attribute name.

### CODE REVIEW

**Comment by michield** [comment]
The build fails on Chrome, but that can be ignored.

We should also try to detect that there's an invalid delimiter used, and either give a warning or suggestion of a different delimiter, but I guess that's slightly more work.

**Comment by bramley** [comment]
I had thought about trying to identify an incorrect field delimiter. I think that the most likely case is someone importing a file with comma or semi-colon separator (i.e. not tab). In this case each line will be treated as just one field. So if the number of columns in the first row of headers is exactly 1 then we could issue a warning when the user is asked to match unidentified column headers.

If there really is only one field in the file then that has to be the email address, so the header column value should be matched automatically and the warning would not be issued.
Maybe something like this

        if (count($headers) == 1) {
            printf('<p class="information">%s</p>', s('The entered field delimiter might not be correct'));
        }


image

**Comment by michield** [comment]
Yes, that's a great idea. Also, we could check on comma and semi-colon, and then add the option to re-submit with the one that was detected. But the suggested change is a good start.

**Comment by bramley** [comment]
I have played around a bit and this seems to work reasonably well. The layout seems a bit cluttered though and the message is a bit wordy.

image

**Comment by michield** [comment]
That's brilliant. I think that will definitely improve the UX here. I think it's a common issue, and I've bumped into it a few times myself. 
This will be a great addition.

File: public_html/lists/admin/import2.php [hunk 1]
```diff
@@ -170,6 +170,13 @@
     $_SESSION['throttle_import'] = !empty($_POST['throttle_import']) ? sprintf('%d', $_POST['throttle_import']) : 0;
 }
 
+if (isset($_GET['delimiter'])) {
+    // Reprocess the file using the selected delimiter
+    $_SESSION['import_field_delimiter'] = $_GET['delimiter'];
```
Comment (michield) [inline public_html/lists/admin/import2.php:175]: This needs sanitising though. You can't just take a get and use it.

**Review by michield — CHANGES_REQUESTED** [review]

**Comment by bramley** [comment]
I have added validation of the GET parameter, and improved the presentation of the message by displaying it within a "note" div.

image

**Comment by michield** [comment]
That's great, thanks. @marianaballa this is ready to merge
